packages/data-explorer-ui/src/components/common/CustomIcon/customIcon.tsx

```diff
@@ -2,7 +2,11 @@ import { SvgIcon, SvgIconProps } from "@mui/material";
 import React from "react";
 import { IconName, IconSvgPathShapes } from "./common/iconSvgPathShapes";
 
-export interface CustomIconProps extends SvgIconProps {
+type SvgIconBaseProps = Pick<
+  SvgIconProps,
+  "color" | "fontSize" | "htmlColor" | "titleAccess" | "ref"
```

Comment:
I limited available fields here to allow only ones that affect icon itself. 

Should we allow something like `sx` or `classNames` by default? I personally always prefer stricter codebase, but happy to add more fields or allow all of them.

Reply:
Yeh, we normally do abide by a stricter code base - I guess in this particular instance, we can't know whether some external application using this component would like to utilize other props like [viewBox](https://mui.com/material-ui/api/svg-icon/).
I think we should probably, for this component, allow the flexibility for all SvgIconProps to be passed through? Thoughts?

Reply:
This component supposed to provide set of predefined icons, that are easy to use.
We don't want users to use the component for any kind of custom icons, they should use SvgIcon itself for such cases. 
So I would certainly not include any kind of svg-related props. 
But styling props -- is something to discuss.

Reply:
Although, thinking this through, currently this component is really set up for package use only (therefore we currently control the set of "custom" icons - let's leave your change as is - we can add viewBox at later date, if needed.